core/iwasm/interpreter/wasm_loader.c

```diff
@@ -9892,6 +9892,15 @@ reserve_block_ret(WASMLoaderContext *loader_ctx, uint8 opcode,
             goto fail;                                                      \
     } while (0)
 
+#define RESERVE_BLOCK_RET_V2()                                              \
+    do {                                                                    \
+        if (!reserve_block_ret(loader_ctx, opcode, disable_emit, error_buf, \
+                               error_buf_size)) {                           \
+            free_label_patch_list(loader_ctx->frame_csp);                   \
```

Comment:
How about move `free_label_patch_list()` to `wasm_loader_ctx_destroy(loader_ctx)`? It seems doable.